### ISSUE TITLE
Possibly misleading error message about moment import

### ISSUE THREAD

**Issue by claremacrae** [obsidianmd/eslint-plugin#143]
### The current recommendation

With this code:

```ts
import type { Moment } from 'moment';
```

I am getting:

```
  2:1  warning  'moment' import is restricted from being used.
The 'moment' package is bundled with Obsidian.
Please import it from 'obsidian' instead  no-restricted-imports
```

### Using the current recommendation (does not compile)

However, when I change it to this:

```ts
import type { Moment } from 'obsidian';
```

I instead get:

```
TS2724: "obsidian" has no exported member named Moment. Did you mean moment?
```

### What actually worked

After a long dance, I actually ended up with this working:

```diff
Index: src/DateTime/DateFallback.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/DateTime/DateFallback.ts b/src/DateTime/DateFallback.ts
--- a/src/DateTime/DateFallback.ts	(revision 4abc3e271daf676784c621de7e523cddd9089931)
+++ b/src/DateTime/DateFallback.ts	(date 1777984089882)
@@ -1,8 +1,9 @@
-import type { Moment } from 'moment';
 import { getSettings } from '../Config/Settings';
 import type { TaskLocation } from '../Task/TaskLocation';
 import { Task } from '../Task/Task';
 
+type Moment = moment.Moment;
+
 /**
  * Implement date from path detection
  */
```

### Improving the error message

So I think the error message should be adjusted to convey that:

1. Obsidian makes `moment` available with no imports, so you may be able to just delete your import line.
2. But to reduce the required changes, you may wish to add this after your imports.:

```ts
type Moment = moment.Moment;
```

**Comment by claremacrae** [comment]
Unfortunately the above `type Moment = moment.Moment;` in a `global.d.ts` is giving me a load of:

```
/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/src/DateTime/DateFallback.ts
  15:43  warning  'Moment' is not defined  no-undef
  42:55  warning  'Moment' is not defined  no-undef
  77:20  warning  'Moment' is not defined  no-undef
  78:24  warning  'Moment' is not defined  no-undef
  79:18  warning  'Moment' is not defined  no-undef
  90:87  warning  'Moment' is not defined  no-undef
```

So I'm no closer to getting this working.

**Comment by rzbin** [comment]
I had a similar issue when using
```ts
import { moment } from "obsidian"
```
giving the error
```
This expression is not callable.
  Type 'typeof moment' has no call signatures.
obsidian.d.ts(8, 1): Type originates at this import. A namespace-style import cannot be called or constructed, and will cause a failure at runtime. Consider using a default import or import require here instead.
```
It seems `obsidian.d.ts` exports the namespace but not the callable function.
What ended up working for me without losing points on the scorecard was using `window.moment()` instead of `moment()`, which does not need any imports.

**Comment by claremacrae** [comment]
@rzbin, we may be thinking of different things - I'm trying to access the type, not the function.

**Comment by claremacrae** [comment]
Many thanks!
